src/epik_eval/__main__.py

```diff
@@ -0,0 +1,483 @@
+                pad_token_id=eval_tokenizer.pad_token_id,
+                synced_gpus=args.synced_gpus)
+        
+        results.append((prompt_id, output, nonduplicate_mask))
+
+        # Check if main process is finished iterating questions
+        if accelerator.gather(torch.tensor(eval_dataloader.dataset.finished, device=device)).any():
+            break
+
+    results = accelerator.pad_across_processes(
+        results,
+        dim=1,
+        pad_index=eval_tokenizer.pad_token_id,
+        pad_first=eval_tokenizer.padding_side == 'left')
+    results = accelerator.gather(results)
+
+    # Logging    
+    if accelerator.is_main_process:
+        correct = [0] * len(eval_dataloader.dataset.targets)
+        model_answers = [None] * len(eval_dataloader.dataset.targets)
+        for prompt_id, output, nonduplicate_mask in results:
+            prompt_id, output = prompt_id[nonduplicate_mask], output[nonduplicate_mask]
+            non_padding = output != eval_tokenizer.pad_token_id
+            for i in range(output.shape[0]):
+                # Remove padding, bos_token if present and eos_token (always present)
+                model_answer = eval_tokenizer.decode(output[i, non_padding[i]][1 if eval_tokenizer.bos_token_id is not None else 0: -1])
+                
+                if args.model_type == 'causal':
+                    # remove the prompt part and the extra character between the prompt and the answer
+                    prompt_len = len(eval_dataloader.dataset.prompts[prompt_id[i]]) + 1
+                    model_answers[prompt_id[i]] = model_answer[prompt_len:]
+                elif args.model_type == 't5':
+                    model_answers[prompt_id[i]] = model_answer
+
+                target_answer = eval_dataloader.dataset.targets[prompt_id[i]]
+                if model_answers[prompt_id[i]][:len(target_answer)] == target_answer:
+                    correct[prompt_id[i]] = 1
+                else:
+                    correct[prompt_id[i]] = 0
+
+        # Log model answers
+        model_answers_csv = eval_dataloader.dataset.eval_df.copy()
+        model_answers_csv.rename(columns={'answer': 'target answer'}, inplace=True)
+        model_answers_csv['model answer'] = model_answers
+        model_answers_csv['correct'] = correct
+
+        accuracies = {}
+        for set in ['train', 'val', 'test']:
+            questions = model_answers_csv[model_answers_csv['set'] == set]
+            accuracies[set] = 100 * questions['correct'].sum() / len(questions['correct'])
+            acc_per_task = {}
+            for task in questions['task'].unique():
+                task_questions = questions[questions['task'] == task]
+                acc_per_task[f'{set.capitalize()} Task {task} Accuracy'] = 100 * task_questions['correct'].sum() / len(task_questions['correct'])
+            accuracies[f'{set}_per_task'] = acc_per_task
+
+        csv_path = args.model_answers_csv.replace('.csv', f'_epoch{epoch}.csv')
```

Comment:
Minor: since you are already using path lib, you can use `path.with_stem(f"{path.stem}_epoch{epoch}")`.